### ISSUE TITLE
Why can’t I pass an authorization header?

### ISSUE THREAD

**Issue by mauricev** [flutterchina/dio#1614]
### New Issue Checklist

- [ X] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   iOS   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |  16.2    |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |     4.0.6 |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |    Android Studio  2021.3.1 Patch 1 |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

I am having trouble getting Dio to pass an authorization header to the web server. From what I've seen here, typical calls are as follows

```
Options? options = Options(headers: { 'authorization':  'Bearer ${session_key}', });
```
For whatever reason, this is not getting passed to the web server SERVER globals. I tried alternative methods of passing like so

```
Map<String, dynamic> header = {
      'authorization': 'Bearer ${session_key}',
      };

    final _options2 = BaseOptions(
      baseUrl: 'http://192.168.1.98:8080/',
      connectTimeout: 5000,
      receiveTimeout: 3000,
      contentType: Headers.formUrlEncodedContentType,
      headers:  header,
    );
```
But it doesn’t pass either way.

For get requests at least, the QUERY_STRING does pass, so something is getting through.

**Comment by mauricev** [comment]
Dio is, in fact, passing the header. The problem is that Apache is not passing it down to php. Putting these lines into the Apache config fixes that:

```
LoadModule rewrite_module lib/httpd/modules/mod_rewrite.so
RewriteEngine On
RewriteRule .* - [e=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
```
